Kernel/Storage/NVMe/NVMeQueue.h

```diff
@@ -44,36 +52,50 @@ class NVMeQueue : public AtomicRefCounted<NVMeQueue> {
     }
     NVMeQueue(NonnullOwnPtr<Memory::Region> rw_dma_region, Memory::PhysicalPage const& rw_dma_page, u16 qid, u32 q_depth, OwnPtr<Memory::Region> cq_dma_region, Vector<NonnullRefPtr<Memory::PhysicalPage>> cq_dma_page, OwnPtr<Memory::Region> sq_dma_region, Vector<NonnullRefPtr<Memory::PhysicalPage>> sq_dma_page, Memory::TypedMapping<DoorbellRegister volatile> db_regs);
 
+    [[nodiscard]] u32 get_request_cid()
+    {
+        u32 expected_tag = m_tag.load(AK::memory_order_acquire);
+
+        for (;;) {
+            u32 cid = expected_tag + 1;
+            if (cid == m_qdepth)
+                cid = 0;
+            if (m_tag.compare_exchange_strong(expected_tag, cid, AK::memory_order_acquire))
```

Comment:
I think with a compare exchange strong, you don't need the loop b/c it can't spuriously fail.

Reply:
Right. As I need to wrap the variable, isn't it better if it is in a loop so that I don't have to repeat the same calculation again for the failure case where another thread updated the value before the current thread could finish executing?